lib/secrets.rb

```diff
@@ -82,6 +87,33 @@ def write_secrets
     @logger.info(message: "secrets written")
   end
 
+  def write_pki_certs
+    errors = []
+    pkis = @pki_keys.map do |name, path|
+      begin
+        uri_path, data = params_from_path(path)
+        [name, write_to_vault(uri_path, data)]
+      rescue StandardError
+        errors << $!
+      end
+    end
+
+    present_errors(errors)
+
+    pkis.each do |name, data|
+      cert_dir = "#{@output_path}/pki/#{name}"
+      FileUtils.mkdir_p cert_dir
+      data.map do |key, value|
+        ext = value.to_s.start_with?(PEM_PREFIX) ? '.pem' : ''
+        file_path = "#{cert_dir}/#{key}#{ext}"
+        @logger.info(message: "Writing PKI object to #{file_path}")
```

Comment:
the idea is to separate message and values to improve searchability
```suggestion
        @logger.info(message: "Writing PKI object", file: file_path)
```